ethcore/src/state_db.rs

```diff
@@ -125,14 +160,103 @@ impl StateDB {
  			try!(Self::commit_bloom(batch, bloom_lock.drain_journal()));
  		}
 		let records = try!(self.db.commit(batch, now, id, end));
-		if self.is_canon {
-			self.commit_cache();
-		} else {
-			self.clear_cache();
-		}
+		self.commit_hash = Some(id.clone());
+		self.commit_number = Some(now);
 		Ok(records)
 	}
 
+	/// Update canonical cache. This should be called after the block has been commited and the
```

Comment:
"Update canonical cache" not a particularly enlightening description of what this mammoth function is meant to be doing. A lot more detail needed. What is its purpose and how does it achieve that?